kf/bpf.go

```diff
@@ -1005,74 +934,62 @@ func (b *BPF) RemovePrevProgFD() error {
 
 		for i := 0; i < 10; i++ {
-			if _, err = os.Stat(b.MapNamePath); err == nil {
-				log.Info().Msgf("VerifyPinnedMapExists : map file created %s", b.MapNamePath)
+			_, err = os.Stat(b.MapNamePath)
+			if err == nil && exists {
+				log.Info().Msgf("VerifyPinnedProgMap creation : map file created %s", b.MapNamePath)
 				return nil
+			} else if err != nil && !exists {
+				if _, err = os.Stat(b.MapNamePath); os.IsNotExist(err) {
+					log.Info().Msgf("VerifyPinnedProgMap removal : map file removed successfully - %s ", b.MapNamePath)
+					return nil
+				} else if err != nil {
+					log.Warn().Err(err).Msg("VerifyPinnedProgMap removal : Error checking for map file")
+				} else {
+					log.Warn().Msg("VerifyPinnedProgMap removal : program pinned file still exists, checking again after a second")
+				}
 			}
-			log.Warn().Msgf("failed to find pinned file, checking again after a second ... ")
 			time.Sleep(1 * time.Second)
 		}
 
-		if err != nil {
-			err = fmt.Errorf("failed to find pinned file %s err %v", b.MapNamePath, err)
+		if err != nil && exists {
+			err = fmt.Errorf("VerifyPinnedProgMap creation : failed to find pinned file %s err %v", b.MapNamePath, err)
 			log.Error().Err(err).Msg("")
-			return err
-		}
-	}
-
-	return nil
-}
-
-// VerifyPinnedMapVanish - making sure XDP program fd map's pinned file is removed
-func (b *BPF) VerifyPinnedMapVanish(chain bool) error {
-
-	if len(b.Program.MapName) <= 0 || b.Program.ProgType != models.XDPType || !chain {
-		return nil
-	}
-
-	var err error
-	log.Debug().Msgf("VerifyPinnedMapVanish : Program %s MapName %s", b.Program.Name, b.Program.MapName)
-
-	for i := 0; i < 10; i++ {
-		if _, err = os.Stat(b.MapNamePath); os.IsNotExist(err) {
-			log.Info().Msgf("VerifyPinnedMapVanish : map file removed successfully - %s ", b.MapNamePath)
-			return nil
 		} else if err != nil {
-			log.Warn().Err(err).Msg("VerifyPinnedMapVanish: Error checking for map file")
-		} else {
-			log.Warn().Msg("VerifyPinnedMapVanish: program pinned file still exists, checking again after a second")
+			err = fmt.Errorf("VerifyPinnedProgMap removal : %s map file was never removed by BPF program %s err %v", b.MapNamePath, b.Program.Name, err)
+			log.Error().Err(err).Msg("")
 		}
-		time.Sleep(1 * time.Second)
+		return err
 	}
 
-	err = fmt.Errorf("%s map file was never removed by BPF program %s err %v", b.MapNamePath, b.Program.Name, err)
-	log.Error().Err(err).Msg("")
-	return err
+	return nil
 }
 
-// This method to verify cmd and process object is populated or not
+// VerifyProcessObject - This method to verify cmd and process object is populated or not
 func (b *BPF) VerifyProcessObject() error {
 
 	if b.Cmd == nil {
 		err := fmt.Errorf("command object is nil - %s", b.Program.Name)
-		log.Error().Err(err).Msg("")
+		log.Error().Err(err).Msg("command object is nil -")
 		return err
 	}
 
 	for i := 0; i < 10; i++ {
```

Comment:
what is the point of this loop?  It doesn't have any sleep time, so not sure why we are running the same check 10 times without delay.